lib/Bindings/Python/dialects/om.py

```diff
@@ -0,0 +1,89 @@
+#  Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+#  See https://llvm.org/LICENSE.txt for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from ._om_ops_gen import *
+from .._mlir_libs._circt._om import Evaluator as BaseEvaluator, Object
+
+from circt.ir import Attribute, Diagnostic, DiagnosticSeverity, Module, StringAttr
+from circt.support import attribute_to_var, var_to_attribute
+
+import sys
+import logging
+from dataclasses import fields
+from typing import TYPE_CHECKING, Any, TypeVar
+
+if TYPE_CHECKING:
+  from _typeshed.stdlib.dataclass import DataclassInstance
+
+
+# Define the Evaluator class by inheriting from the base implementation in C++.
+class Evaluator(BaseEvaluator):
+
+  def __init__(self, mod: Module) -> None:
+    """Instantiate an Evaluator with a Module."""
+
+    # Call the base constructor.
+    super().__init__(mod)
+
+    # Set up logging for diagnostics.
+    logging.basicConfig(
+        format="[%(asctime)s] %(name)s (%(levelname)s) %(message)s",
+        datefmt="%Y-%m-%d %H:%M:%S",
+        level=logging.INFO,
+        stream=sys.stdout,
+    )
+    self._logger = logging.getLogger("Evaluator")
+
+    # Attach our Diagnostic handler.
+    mod.context.attach_diagnostic_handler(self._handle_diagnostic)
+
+  def instantiate(self, cls: type["DataclassInstance"],
```

Comment:
type is subscriptable only since python 3.9 [PEP-585](https://peps.python.org/pep-0585/) so the current implementation will cause import error in python <= 3.8. Perhaps we can use `typing.Type`?

Reply:
Good call, we should be able to use `typing.Type`. I'll also check if `from __future__ import annotations` is sufficient, since we have been using that, which is available from 3.7 onwards. I think 3.7 is the oldest we've ever supported, since 3.6 is already EOL and 3.7 will be in the next months.

Reply:
Looks like `from __future__ import annotations` works for subscripting `type` in 3.7 onward. Since we're already using that pattern, I'll use that here.